docs/src/userguide/debugging.rst

```diff
@@ -9,19 +9,15 @@ Debugging your Cython program
 Cython comes with an extension for the GNU Debugger that helps users debug
 Cython code. To use this functionality, you will need to install gdb 7.2 or
 higher, built with Python support (linked to Python 2.6 or higher).
-The debugger supports debuggees with versions 2.6 and higher. For Python 3,
-code should be built with Python 3 and the debugger should be run with
-Python 2 (or at least it should be able to find the Python 2 Cython
-installation). Note that in recent versions of Ubuntu, for instance, ``gdb``
-installed with ``apt-get`` is configured with Python 3. On such systems, the
-proper configuration of ``gdb`` can be obtained by downloading the ``gdb``
-source, and then running::
-
-    ./configure --with-python=python2
+For optimal results, you should configure gdb from source and use the --with-python
+configuration flag. ::
```

Comment:
It feels like we could still say a bit more about ways to get this working with recent Python 3 deployments. Most Linux installations these days might actually work out of the box with gdb and Python 3. Users shouldn't have to build anything as basic as gdb from sources.

Reply:
Yea I couldn't find good info on this, but as far as I could tell gdb needed to be compiled with a reference to the Python installation you want to use to debug. I wasn't able to use a virtual environment and get it to work

Reply:
OK it may not be as strict as this documentation shows, but it is not as easy as using the system gdb unfortunately. It seems like when calling a system-installed gdb from a virtual environment like conda, gdb loses track of python being called from a local environment and falls back to the system python

```sh
(pandas-dev) willayd@willayd-Lemur-Pro:~/clones/pandas$ which python
/home/willayd/mambaforge/envs/pandas-dev/bin/python
(pandas-dev) willayd@willayd-Lemur-Pro:~/clones/pandas$ cygdb
GNU gdb (Ubuntu 14.0.50.20230907-0ubuntu1) 14.0.50.20230907-git
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
There was an error in Python code originating from the file /home/willayd/clones/cython/Cython/Debugger/Cygdb.py
It used the Python interpreter /usr/bin/python
Traceback (most recent call last):
  File "<string>", line 11, in <module>
ModuleNotFoundError: No module named 'Cython'
```

Installing gdb into the virtual environment resolves the issue. Not sure yet if this is a Cython or gdb problem mixing up the python installations

Reply:
Looking at the gdb source code there is also some compile-time compatability for versions prior to 3.9. This particular instance probably just deserves a note in the documentation, but there is a risk for that happening in the future 

https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=gdb/python/python.c;h=8a36673a3e4c0bef8533d259d8c13118e650df1a;hb=refs/heads/master#l2213